crates/exex/src/atlantic/endpoints.rs

```diff
@@ -0,0 +1,266 @@
+    pub trace_generation: Url,
+    /// Submit an Atlantic trace generation and proof generation query
+    pub trace_generation_proof_generation: Url,
+    /// Submit a L2 Atlantic query
+    pub atlantic_query: Url,
+    /// Submit a L2 Atlantic proof generation and proof verification query
+    pub proof_generation_verification: Url,
+    /// Submit a L2 Atlantic proof verification query
+    pub proof_verification: Url,
+}
+
+impl L2Endpoints {
+    pub fn new(base_url: &Url) -> Result<Self, SharpSdkError> {
+        Ok(Self {
+            proof_generation: base_url.join("proof-generation")?,
+            trace_generation: base_url.join("trace-generation")?,
+            trace_generation_proof_generation: base_url
+                .join("trace-generation-proof-generation")?,
+            atlantic_query: base_url.join("l2/atlantic-query")?,
+            proof_generation_verification: base_url
+                .join("l2/atlantic-query/proof-generation-verification")?,
+            proof_verification: base_url.join("l2/atlantic-query/proof-verification")?,
+        })
+    }
+}
+
+/// The Sharp queries endpoints.
+#[derive(Debug, Clone)]
+pub struct SharpQueriesEndpoints {
+    /// Get the list of Atlantic queries submitted by the user
+    pub get_queries: Url,
+    /// Get the details of a specific Atlantic query
+    pub get_query: Url,
+    /// Get the list of jobs for a specific Atlantic query
+    pub get_query_jobs: Url,
+}
+
+impl SharpQueriesEndpoints {
+    pub fn new(base_url: &Url) -> Result<Self, SharpSdkError> {
+        Ok(Self {
+            get_queries: base_url.join("atlantic-queries")?,
+            get_query: base_url.join("atlantic-query")?,
+            get_query_jobs: base_url.join("atlantic-query-jobs")?,
+        })
+    }
+}
+
+/// The health check endpoint.
+#[derive(Debug, Clone)]
+pub struct HealthCheckEndpoint {
+    /// Check if the server is alive
+    pub is_alive: Url,
+}
+
+impl HealthCheckEndpoint {
+    pub fn new(base_url: &Url) -> Result<Self, SharpSdkError> {
+        Ok(Self { is_alive: base_url.join("is-alive")? })
+    }
+}
+
+/// The program registry endpoint.
+#[derive(Debug, Clone)]
+pub struct ProgramRegistryEndpoint {
+    /// Submit a program to the program registry
+    pub submit_program: Url,
+}
+
+impl ProgramRegistryEndpoint {
+    pub fn new(base_url: &Url) -> Result<Self, SharpSdkError> {
+        Ok(Self { submit_program: base_url.join("submit-program")? })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_l2_endpoints_invalid_base_url() {
+        // Test with an invalid base URL
+        let base_url = Url::parse("invalid-url");
+        assert!(base_url.is_err(), "Expected base URL parsing to fail");
+    }
+
+    #[test]
+    fn test_l2_endpoints_valid_base_url_trailing_slash() {
+        let base_url = Url::parse("https://example.com/")
+            .expect("Failed to parse base URL with trailing slash");
```

Comment:
again here, external deps are supposed to work as expected, we can just `.?` or `unwrap`

Reply:
here this is exactly the same as unwrap in rust just with a nice message in case of failure but the behaviour is exactly the same

Reply:
but it creates more words to read and gives the impression that we expect this to possibly fail, which we don't want to bother at this place

Reply:
fixed